src/MainWindow.vala

```diff
@@ -847,15 +848,50 @@ namespace Terminal {
             }
         }
 
-        private void update_context_menu () {
+        public void update_context_menu () {
+            /* Update the "Paste" menu option */
             clipboard.request_targets (update_context_menu_cb);
+
+            /* Update the "Show in ..." menu option */
+            get_current_selection_link_or_pwd ((clipboard, uri) => {
+                update_menu_label (strip_uri (uri));
+            });
+        }
+
+        private void update_menu_label (string? uri) {
+            AppInfo? appinfo = null;
+            var scheme = uri != null ? Uri.parse_scheme (uri) : "";
+            if (scheme != null) {
+                appinfo = AppInfo.get_default_for_uri_scheme (scheme);
+            }
+
+            if (appinfo == null && uri != null && uri.contains (Path.DIR_SEPARATOR_S)) {
+                bool uncertain;
+                /* Guess content type from filename if possible */
+                //TODO Get content type from actual file (if exists)
+                var ctype = ContentType.guess (uri, null, out uncertain);
+                if (!uncertain) {
+                    appinfo = AppInfo.get_default_for_type (ctype, true);
+                }
+
+                if (appinfo == null) {
+                    /* Fallback to File Browser */
+                    appinfo = AppInfo.get_default_for_type ("inode/directory", true);
+                }
+            }
```

Comment:
Can you explain what case this covers? I'm unsure what this is intended to do

Reply:
This covers everything that 
* does not have an explicit scheme and/or
* Does not have a recognisable extension (i.e. the content type cannot be guessed)

For example, the pwd is opened through this fallback.  This is a simple, low code alternative to creating a GLib.File and trying to read the file attributes (which is what Files does).  I am not sure of what use cases people have and therefore how sophisticated it is worth making this.

Reply:
After latest commits, no longer guess content.  Also interpret "~" and ".." in path. Needs testing with unusual characters in filename.